kyo-core/shared/src/main/scala/kyo/Duration.scala

```diff
@@ -0,0 +1,74 @@
+package kyo
+
+import java.time.Duration as JDuration
+import java.time.temporal.ChronoUnit.*
+import java.time.temporal.TemporalUnit
+import scala.annotation.switch
+import scala.concurrent.duration.Duration as SDuration
+
+opaque type Duration <: JDuration = JDuration
```

Comment:
Do you think we could use `opaque type Duration = Long`? It'd be nice to avoid the allocation of Java's wrapper.

Reply:
It's possible, but my goal was to directly 'replace' Java Duration. Can we make it Java duration compatible - ie, a value class that extends `Duration`?

Reply:
I don't think so, value classes can only extend "universal traits": https://docs.scala-lang.org/overviews/core/value-classes.html.

Maybe a `toJava` would be enough to provide interop?